### PR TITLE
Change VGG19 example to tf.layers.

### PR DESCRIPTION
**Pull request by PatWie**
As part of the transition towards tf.layers #291, #627 this is a
rewrite. While The LinearWrap might need less lines, tf.layers are
supported by TensorFlow. And the motivation of tensorpack is being
a training framework.

### CODE REVIEW

File: examples/CaffeModels/load-vgg19.py [hunk 1]
```diff
@@ -7,53 +7,58 @@
...
-                  .FullyConnected('fc8', 1000)())
+    with argscope([tf.layers.conv2d], kernel_size=3, activation=tf.nn.relu, padding='same'):
+        x = image
+        x = tf.layers.conv2d(x, 64, name='conv1_1')
+        x = tf.layers.conv2d(x, 64, name='conv1_2')
+        x = tf.layers.max_pooling2d(x, 2, 2, name='pool1')
+
+        x = tf.layers.conv2d(x, 128, name='conv2_1')
+        x = tf.layers.conv2d(x, 128, name='conv2_2')
+        x = tf.layers.max_pooling2d(x, 2, 2, name='pool1')
```
Comment (ppwwyyxx) [inline examples/CaffeModels/load-vgg19.py]: should better be 'pool2'

File: examples/CaffeModels/load-vgg19.py [hunk 2]
```diff
@@ -7,53 +7,58 @@
...
+
+        x = tf.layers.conv2d(x, 512, name='conv5_1')
+        x = tf.layers.conv2d(x, 512, name='conv5_2')
+        x = tf.layers.conv2d(x, 512, name='conv5_3')
+        x = tf.layers.conv2d(x, 512, name='conv5_4')
+        x = tf.layers.max_pooling2d(x, 2, 2, name='pool1')
+        x = tf.layers.flatten(x, name='flatten')
+
+        x = tf.layers.dense(x, 4096, activation=tf.nn.relu, name='fc6')
+        x = tf.layers.dropout(x, rate=0.5, name='drop0')
```
Comment (ppwwyyxx) [inline examples/CaffeModels/load-vgg19.py]: This will actually do the dropout? Tensorpack's dropout layer knows the context is inference, and will not perform the dropout.

Comment (PatWie) [reply examples/CaffeModels/load-vgg19.py]: I always test the implementation before doing a PR. The default is inference-mode for `tf.layers.dropout` (training=False).

File: examples/CaffeModels/load-vgg19.py (hunk 1 above)
Comment (PatWie) [reply examples/CaffeModels/load-vgg19.py]: You are right! Done in the next commit.
